### PR TITLE
Fix pincode on Android

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

**Comment by megrogan** [comment]
I will make this better...

File: frontend/app/src/components/pincode/Pincode.svelte [hunk 1]
```diff
@@ -0,0 +1,189 @@
+<script>
+    /**
```
Comment (julianjelfs) [inline frontend/app/src/components/pincode/Pincode.svelte:2]: Pretty surprised these got through all the linting! 

Would be nice to convert these to typescript and scss.

File: frontend/app/src/components/pincode/PincodeInput.svelte [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            return;
+        }
+
+        if (e.key !== KEYBOARD.TAB) {
+            e.preventDefault();
+        }
+
+        // Do not try to update the value from the keydown event if on android, leave that to the input event
+        if (!android) {
+            if (type === "numeric" && /^[0-9]$/.test(e.key)) {
```
Comment (megrogan) [inline frontend/app/src/components/pincode/PincodeInput.svelte:98]: Would be good to see this code block shared by keydown/input

File: frontend/app/src/components/pincode/PincodeInput.svelte [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    export let value: string = "";
+    export let id: string = "input" + Math.random().toString(36);
+
+    let ref: HTMLInputElement | undefined;
+    let type: PincodeType;
+    let selectTextOnFocus: boolean;
+    let userAgent = navigator.userAgent;
+
+    const android = userAgent?.match(/android/i);
```
Comment (megrogan) [inline frontend/app/src/components/pincode/PincodeInput.svelte:13]: Is there a capability based way of determining whether keydown will work?

Comment (julianjelfs) [reply frontend/app/src/components/pincode/PincodeInput.svelte:14]: Not really sure that there is.

**Review by julianjelfs — APPROVED** [review]

**Review by julianjelfs — APPROVED** [review]

**Pull request merged**